pkg/apis/networking/validation/validation.go

```diff
@@ -597,3 +597,107 @@ func allowInvalidWildcardHostRule(oldIngress *networking.Ingress) bool {
+func ValidateClusterCIDRConfigSpec(spec *networking.ClusterCIDRConfigSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if spec.NodeSelector != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNodeSelector(spec.NodeSelector, fldPath.Child("nodeSelector"))...)
+	}
+
+	maxIPv4PerNodeMaskSize := int32(32)
+	maxIPv6PerNodeMaskSize := int32(128)
+
+	hasIPv4 := spec.IPv4 != nil
+	hasIPv6 := spec.IPv6 != nil
+
+	// Validate if CIDR is configured for at least one IP Family(IPv4/IPv6).
+	if !hasIPv4 && !hasIPv6 {
+		allErrs = append(allErrs, field.Required(fldPath, "one or both of `ipv4` and `ipv6` must be configured"))
+	}
+
+	// Validate configured IPv4 CIDR and PerNodeMaskSize.
+	if hasIPv4 {
+		allErrs = append(allErrs, validateCIDRConfig(spec.IPv4, fldPath, maxIPv4PerNodeMaskSize, "ipv4")...)
+	}
+
+	// Validate configured IPv6 CIDR and PerNodeMaskSize.
+	if hasIPv6 {
+		allErrs = append(allErrs, validateCIDRConfig(spec.IPv6, fldPath, maxIPv6PerNodeMaskSize, "ipv6")...)
+	}
+
+	// Validate PerNodeMaskSize if both IPv4 and IPv6 CIDRs are configured.
+	// IPv4.PerNodeMaskSize and IPv6.PerNodeMaskSize must specify the same number of IP addresses:
+	// 32 - spec.IPv4.PerNodeMaskSize == 128 - spec.IPv6.PerNodeMaskSize
+	// Unequal allocatable IP addresses will lead to wastage of IP addresses.
+	// Also, this being one of the tie-breaker rules for ClusterCIDConfigs, it is
+	// necessary to have equal number of allocatable IP addresses per node.
+	if hasIPv4 && hasIPv6 {
+		if maxIPv4PerNodeMaskSize-spec.IPv4.PerNodeMaskSize != maxIPv6PerNodeMaskSize-spec.IPv6.PerNodeMaskSize {
+			allErrs = append(allErrs, field.Invalid(fldPath, *spec, "must provide valid PerNodeMaskSize for IPv4 and IPv6, 32 - IPv4.PerNodeMaskSize == 128 - IPv6.PerNodeMaskSize must be satisfied."))
+		}
+	}
+	return allErrs
+}
+
+func validateCIDRConfig(cidrConfig *networking.CIDRConfig, fldPath *field.Path, maxPerNodeMaskSize int32, ipFamily string) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if cidrConfig.CIDR == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child(ipFamily, "cidr"), "must provide a CIDR"))
+	} else {
+		switch ipFamily {
+		case "ipv4":
+			if !netutils.IsIPv4CIDRString(cidrConfig.CIDR) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child(ipFamily, "cidr"), *cidrConfig, "must be a valid IPv4 CIDR"))
+			}
+		case "ipv6":
+			if !netutils.IsIPv6CIDRString(cidrConfig.CIDR) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child(ipFamily, "cidr"), *cidrConfig, "must be a valid IPv6 CIDR"))
+			}
+		}
+	}
+
+	if cidrConfig.PerNodeMaskSize == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child(ipFamily, "perNodeMaskSize"), fmt.Sprintf("must provide an %s PerNodeMaskSize", ipFamily)))
+	} else {
+		if _, cidr, err := netutils.ParseCIDRSloppy(cidrConfig.CIDR); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child(ipFamily, "cidr"), *cidrConfig, "must be a valid CIDR"))
```

Comment:
You already checked this above - if you can't verify it as the correct family just skip this check - it's going to reject anyway.

Reply:
removed the error check from here